src/Text/Layout/Table/Cell/WideString.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiWayIf                 #-}
+{-# LANGUAGE OverloadedStrings          #-}
+module Text.Layout.Table.Cell.WideString
+    ( WideString(..)
+    , WideText(..)
+    ) where
+
+import Data.String
+import qualified Data.Text as T
+import Text.DocLayout
+
+import Text.Layout.Table.Cell
+import Text.Layout.Table.Primitives.AlignInfo
+import Text.Layout.Table.StringBuilder
+
+-- | A newtype for String in which characters can be wider than one space.
+newtype WideString = WideString String
+    deriving (Eq, Ord, Show, Read, Semigroup, Monoid, IsString, StringBuilder)
+
+instance Cell WideString where
+    dropLeft i (WideString s) = WideString $ dropWide True i s
+    dropRight i (WideString s) = WideString . reverse . dropWide False i $ reverse s
+    visibleLength (WideString s) = realLength s
+    measureAlignment p (WideString s) = measureAlignmentWide p s
+
+    buildCell (WideString s) = buildCell s
+
+dropWide :: Bool -> Int -> String -> String
+dropWide _ i [] = []
+dropWide gobbleZeroWidth i (x:xs)
+    | gobbleZeroWidth && i == 0 && l == 0 = dropWide gobbleZeroWidth i xs
+    | i <= 0    = x:xs
```

Comment:
Please use spaces around operators and also in patterns.  Actually, I would prefer if you could use a named pattern and use it here instead of constructing a new list.
```
dropWide gobbleZeroWidth i l@(x : xs)
...
    | i <= 0     = l
```